projects/step-frontend/src/lib/modules/_common/components/sidebar/sidebar.component.html

```diff
@@ -0,0 +1,63 @@
+<div id="sidebar-logo-and-burgermenu">
+  <span class="visible-when-collapsed"
+    ><label for="collapsable-sidebar-toggle-checkbox"><i class="glyphicon glyphicon-menu-hamburger"></i></label
+  ></span>
+  <span class="sidebar-aligned-text"><img class="sidebar-logo" alt="" [src]="logo" /></span>
+</div>
+<div id="sidebar-home-and-settings">
+  <span class="visible-when-collapsed sidebar-home-button" (click)="navigateTo('')"
+    ><i class="glyphicon glyphicon-home"></i
+  ></span>
+  <span class="sidebar-home-and-settings-divider">|</span>
+  <span (click)="navigateTo('admin')" *ngIf="_authService.hasRight('admin')" class="sidebar-home-and-settings-icon"
+    ><i class="glyphicon glyphicon-cog"></i
+  ></span>
+</div>
+<div id="sidebar-categories">
+  <div class="sidebar-categories-categories">
+    <span
+      (click)="navigateTo('admin')"
+      *ngIf="_authService.hasRight('admin')"
+      class="only-visible-when-collapsed sidebar-home-and-settings-icon-collapsed"
+      style="font-size: medium"
+      ><i class="glyphicon glyphicon-cog"></i
+    ></span>
+    <span>Categories</span>
+  </div>
+  <div id="sidebar-categories-tabs">
+    <!-- MENU -->
+    <div
+      class="sidebar-collapsable-tab"
+      *ngFor="let mme of _viewRegistryService.getCustomMainMenuEntries(); index as i"
```

Comment:
This is a huge problem, you should never invoke functions from within ng2+ html code. This function will be called on every tick and is a mssive resource leak even if it just returns an array. The solution for this is using pipes.

Reply:
absolutely, I just sketched this in and forgot to fix it afterwards, probably I would have noticed it next time I would have seen it. Thanks for pointing it out. The most simple solution here is not a pipe but simply calling this method once in the constructor and save the result, since the entries will not change after the application has started.